File: 404_sum_of_left_leaves/check_left_leaf.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum_l_leaves = 0;
    void traversal(TreeNode* curr)
    {
        if(curr == NULL)
            return;

        if(curr->left != NULL)
        {
            if(curr->left->left == NULL && curr->left->right == NULL)
                sum_l_leaves += curr->left->val;
        }

        traversal(curr->left);
        traversal(curr->right);
    }
    int sumOfLeftLeaves(TreeNode* root) {
        traversal(root);
        return sum_l_leaves;
    }
};

File: 404_sum_of_left_leaves/check_left_leaf.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    sum_l_leaves = 0
    def traversal(self, curr):
        if curr == None:
            return

        if curr.left != None:
            if curr.left.left == None and curr.left.right == None:
                self.sum_l_leaves += curr.left.val

        self.traversal(curr.left)
        self.traversal(curr.right)

    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        self.traversal(root)
        return self.sum_l_leaves